x/distribution/module.go

```diff
@@ -16,6 +17,54 @@ type Module struct {
 func (m Module) Initialize(client module.Client) module.Descriptor {
 	return module.NewDescriptorBuilder().
 		Named("distribution").
-		ExtendsAuthentication(extensions2.NewAuthentication(client)).
+		WithAuthAdmissionHandler(NewFeeChecker(client)).
+		WithPostAuthenticationHandler(NewFeeDeduction(client)).
 		NeedsStateTransition(&v1alpha1.MsgSendCoins{}).Build()
 }
+
+func NewFeeChecker(c module.Client) authentication.AdmissionHandler {
```

Comment:
same as above, I'd remove this logic from module.go and place it to some other file.